src/archive/archive.did

```diff
@@ -0,0 +1,143 @@
+type HeaderField = record { text; text; };
+
+type HttpRequest = record {
+  method: text;
+  url: text;
+  headers: vec HeaderField;
+  body: blob;
+};
+
+type HttpResponse = record {
+  status_code: nat16;
+  headers: vec HeaderField;
+  body: blob;
+  upgrade : opt bool;
+  streaming_strategy: opt StreamingStrategy;
+};
+
+type StreamingCallbackHttpResponse = record {
+  body: blob;
+  token: opt Token;
+};
+
+type Token = record {};
+
+type StreamingStrategy = variant {
+  Callback: record {
+    callback: func (Token) -> (StreamingCallbackHttpResponse) query;
+    token: Token;
+  };
+};
+
+type Timestamp = nat64;
+type UserNumber = nat64;
+type CredentialId = blob;
+type PublicKey = blob;
+type DeviceKey = PublicKey;
+
+type Purpose = variant {
+    recovery;
+    authentication;
+};
+
+type KeyType = variant {
+    unknown;
+    platform;
+    cross_platform;
+    seed_phrase;
+};
+
+type DeviceProtection = variant {
+    protected;
+    unprotected;
+};
+
+type OperationType = variant {
+    register_anchor: record {
+        initial_device: DeviceDataWithoutAlias;
+    };
+    add_device: record {
+        new_device: DeviceDataWithoutAlias;
+    };
+    update_device: record {
+        updated_device: PublicKey;
+        changed_data: DeviceDataUpdate;
+    };
+    remove_device: record {
+        removed_device: PublicKey;
+    };
+};
+
+type LogEntry = record {
+    user_number: UserNumber;
+    operation: OperationType;
+    timestamp: Timestamp;
+    caller: PublicKey;
+    sequence_number: nat64;
+};
+
+type DeviceDataWithoutAlias = record {
+    pubkey: DeviceKey;
+    credential_id: opt  CredentialId;
+    purpose: Purpose;
+    key_type: KeyType;
+    protection: DeviceProtection;
+};
+
+type DeviceDataUpdate = record {
+    alias: opt Hidden;
+    credential_id: opt CredentialId;
+    purpose: opt Purpose;
+    key_type: opt KeyType;
+    protection: opt DeviceProtection;
+};
+
+// Placeholder for information that has been hidden for privacy reasons.
+type Hidden = variant {
+    hidden_for_privacy_reasons;
+};
+
+type Cursor = variant {
+    timestamp: Timestamp; // get logs starting from this Timestamp
+    next_token: blob; // submit previously received token to fetch the next page of log entries
+};
+
+type UserLogs = record {
+  entries: vec blob;
+  next_token: opt blob // cursor to fetch the next page of log entries (if any)
+};
+
+type Logs = record {
+  entries: vec blob;
+  next_idx: opt nat64 // next index to fetch the next page of log entries (if any)
+};
+
+type ArchiveInit = record {
+  ii_canister: principal;
+};
+
+service : (opt ArchiveInit) -> {
```

Comment:
really nitpicky here, but how about bringing the `service` to the top, or at least hiding all those HTTP types at the button, or making different sections with stuff actually relevant, and stuff needed because candid is what is it?

```

// Our lovely types

type Logs = ...


// Stuff that shouldn't even be here

type AwfulHttpStreamingType = ...
```

Reply:
So far it has been convention that `service` is the very last definition of a candid file (mostly because it requires all previous definitions, at least transitively), Internet Identity does it this way, but also the management canister and all canisters in the IC repo. I would not change this here.

But I agree, that adding a comment on the types is helpful. I will do that.